File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/UIButton_Circle.swift
//
//  UIButton_Circle.swift
//  TelerikUIExamplesInSwift
//
//  Copyright (c) 2015 Telerik. All rights reserved.
//

import UIKit

extension UIButton {

    class func circleButtonInView(view: UIView, title: NSString, target: AnyObject, action: Selector) -> UIButton {
        let button : UIButton = UIButton(type:UIButtonType.Custom)

        button.addTarget(target, action: action, forControlEvents: UIControlEvents.TouchUpInside)
        button.backgroundColor = UIColor(red: 0.5, green: 0.7, blue: 0.2, alpha: 0.7)
        button.setTitleColor(UIColor.whiteColor(), forState: UIControlState.Normal)
        button.setTitle(title as String, forState: UIControlState.Normal)
        button.titleLabel!.font = UIFont(name: "GillSans", size: 14)
        button.layer.cornerRadius = 40
        button.clipsToBounds = true
        button.frame = CGRectMake((CGRectGetWidth(view.frame) - 80)/2.0 , CGRectGetHeight(view.frame)-180, 80, 80)
        
        button.autoresizingMask = UIViewAutoresizing(rawValue:UIViewAutoresizing.FlexibleTopMargin.rawValue | UIViewAutoresizing.FlexibleLeftMargin.rawValue | UIViewAutoresizing.FlexibleRightMargin.rawValue)
        
        view.addSubview(button)
        
        return button
    }
}

File: Examples/Examples/Alert/Classes/UIButton_Circle.cs
using System;
using UIKit;
using CoreGraphics;
using TelerikUI;

namespace Examples
{
	public class UIButton_Circle : UIButton
	{
		public static UIButton Button(UIView View, String Title, Object Target, EventHandler handler)
		{
			UIButton btn = new UIButton (UIButtonType.Custom);
			btn.BackgroundColor = new UIColor (0.5f, 0.7f, 0.2f, 1f);
			btn.SetTitleColor (new UIColor (1f, 1f, 1f, 1f), UIControlState.Normal);
			btn.SetTitle (Title, UIControlState.Normal);
			btn.TitleLabel.Font = UIFont.SystemFontOfSize (12);
			btn.Layer.CornerRadius = 40;
			btn.ClipsToBounds = true;
			btn.Frame = new CGRect ((View.Frame.Size.Width - 80)/2, View.Frame.Size.Height - 180, 80, 80);
			btn.AutoresizingMask = UIViewAutoresizing.FlexibleRightMargin | UIViewAutoresizing.FlexibleTopMargin | UIViewAutoresizing.FlexibleLeftMargin;
			btn.AddTarget(handler, UIControlEvent.TouchUpInside);
			View.AddSubview (btn);
			return btn;
		}
	}
}

